src/topology/local_homeomorph.lean

```diff
@@ -404,6 +407,13 @@ lemma of_set_trans' {s : set α} (hs : is_open s) :
   (of_set s hs).trans e = e.restr (e.source ∩ s) :=
 by rw [of_set_trans, restr_source_inter]
 
+lemma of_set_trans_of_set {s : set α} (hs : is_open s) {s' : set α} (hs' : is_open s') :
+  (of_set s hs).trans (of_set s' hs') = of_set (s ∩ s') (is_open_inter hs hs')  :=
```

Comment:
`simp`?

Reply:
I wasn't sure whether to do this, since its preexisting friends (eg
```
lemma of_set_trans {s : set α} (hs : is_open s) : (of_set s hs).trans e = e.restr s 
```
) are not `simp`.

Reply:
I think `simp` is a good idea here, and also on `of_set_trans`.

Reply:
I've made this lemma `@[simp, mfld_simps]`.  Doing it for the other ones broke some things, for example [this line](https://github.com/leanprover-community/mathlib/blob/d7ac180f24b9c672c028d5bc2094059b4ba72178/src/geometry/manifold/charted_space.lean#L661) in `structure_groupoid.compatible_of_mem_maximal_atlas`, so I did not keep that change.